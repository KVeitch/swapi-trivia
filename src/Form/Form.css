h1 {
  font-family: 'SpaceMeatballRegular';
  font-weight: normal;
  font-style: normal;
  font-size: 4em;
  text-shadow: 4px 3px 0px #fff, 9px 8px 0px rgba(0,0,0,0.15);
}

form {
  display:flex;
  height:100vh;
  font-family: 'Barlow Condensed', sans-serif;
  height: 100vh;
  background: url('https://starwarsblog.starwars.com/wp-content/uploads/2015/07/Screen-Shot-2015-07-06-at-2.58.52-PM.png') no-repeat center center;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  width: 100%;  
}

.sidebar {
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  flex-direction: column;
  height: 100vh;
  background-color: rgba(255, 255, 255, 0.75);
  width: 25%;
}

input, 
select,
option {
  border-radius: .5em;
	border: 1px solid gray;
	box-shadow: 0 1px 0 1px rgba(0,0,0,.04);
  display:block;
  font-size: 1em;
  line-height: 1.3;
	padding: .6em 1.4em .5em .8em;
	-moz-appearance: none;
	-webkit-appearance: none;
  appearance: none;
  margin-top: 15px;
  text-align: center;
  width: 85%;
}

input:focus,
select:focus {
  outline: none;
  box-shadow: 0px 0px 15px red;
}
input::placeholder {
  color:rgba(79, 79, 79, 0.5);
  text-align:center;
  line-height: 5em;
}

.logo {
  height: 15%;
}

.ranking {
  background-color: white;
  border-radius: 5px;
}

.btn__submit {
  font-size: 22px;
  height: 7%;
  width: 42%;
  border-radius: 5%;
  box-shadow: 0 10px 6px -6px rgba(41, 72, 76, 0.99);
  background-color:rgb(223, 213, 207);
  margin-top: 15px;
  cursor: pointer;
}

.quote__submit {
  border-radius: 5%;
  box-shadow: 0 10px 6px -6px rgba(41, 72, 76, 0.99);
  background-color:rgb(223, 213, 207);
  height: 4%;
  font-size: 1em;
  margin-right: 20px;
  width: 15%;
}

.error-icon {
  height:50px
}

.error {
  font-size: 1em;
  height:70px;
  display:flex;
  justify-content: center;
  line-height: 70px;
  background-color: rgb(233, 189, 169);
  border-radius: 5px;
  padding: 15px;
}

.hidden {
  visibility: hidden;
}

.random__quote {
  align-items: flex-end;
  color: white;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  margin-left: auto;
}

.quote__text {
  font-size: 2em;
}
