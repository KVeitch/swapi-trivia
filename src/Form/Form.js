import React from 'react';

class Form {
    constructor() {
        
    }
}

export default Form;