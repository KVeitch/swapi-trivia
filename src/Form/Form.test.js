import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
    let wrapper;
    const mockSetUser = jest.fn();
    const mockSetMovies = jest.fn();
    const mockSetCharacters = jest.fn();
    beforeEach(() => {
        wrapper =
            shallow(<Form
                setUser={mockSetUser}
                setMovies={mockSetMovies}
                setCharacters={mockSetCharacters}
                insults={[{ quote: 'Don’t worry about Master Luke. I’m sure he’ll be all right. He’s quite clever, you know… for a human being.' }]}
            />)
    });

    it('should match the snapshot with all the data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('handleChange', () => {
        it('should update state.name when handleChange is called', () => {
            let mockEvent = {
                target: {
                    name: 'name',
                    value: 'Bob'
                }
            }
            // wrapper.find('.input__name').simulate('onChange');
            wrapper.instance().handleChange(mockEvent);

            expect(wrapper.state().name).toEqual('Bob');
        });

        it('should find the input with a class of .input__name and call the handleChange method', () => {
            let mockEvent = {
                target: {
                    name: 'quote',
                    value: 'Hello World'
                }
            }
            wrapper.find('.input__name').simulate('onChange');
            expect(wrapper.instance().handleChange(mockEvent));
        });

        it('should update state.quote when handleChange is called', () => {
            let mockEvent = {
                target: {
                    name: 'quote',
                    value: 'Hello World'
                }
            }
            // can write a test for wrapper.find finds an input and simulates a change event & calls __()
            // expect ___() to have been called 
            // wrapper.find('.input__quote').simulate('onChange');
            wrapper.instance().handleChange(mockEvent);

            expect(wrapper.state().quote).toEqual('Hello World');
        });
    });

    describe('default values', () => {
        it('should have Nerf Herder as a default value for state.ranking', () => {
            expect(wrapper.state().ranking).toEqual('Nerf Herder');
        });
        it('should have a default value of false for revealError', () => {
            expect(wrapper.state().revealError).toEqual(false);
        });
        it('should have a default value of false for toMovies', () => {
            expect(wrapper.state().toMovies).toEqual(false);
        });
        it('should have default values of empty strings for name, quote, and randomQuote', () => {
            expect(wrapper.state().name).toEqual('');
        });
    });

    describe('handleSubmit', () => {
        it('should be update revealError to true if theres no name or quote', () => {
            expect(wrapper.state().revealError).toEqual(false);
            wrapper.setState({ name: 'Susan' });
            wrapper.instance().handleSubmit();
            expect(wrapper.state().revealError).toEqual(true);
        });

        it('should call setUser if there is a valid name and quote', () => {
            wrapper.setState({ name: 'Susan', quote: "After all, he's a wookie" });
            wrapper.instance().handleSubmit();
            // expect(mockSetUser) also works
            expect(wrapper.instance().props.setUser).toHaveBeenCalledWith('Susan', "After all, he's a wookie", 'Nerf Herder')
        });

        it('should reset state', () => {
            wrapper.setState({ name: 'Bob', ranking: 'Jedi Master', quote: 'Hello World', revealError: true, toMovies: false });
            wrapper.instance().handleSubmit();
            expect(wrapper.state().name).toEqual('');
            expect(wrapper.state().ranking).toEqual('');
            expect(wrapper.state().quote).toEqual('');
            expect(wrapper.state().revealError).toEqual(false);
            expect(wrapper.state().toMovies).toEqual(true);
        });
    });

    describe('handleRandomQuote', () => {
        it('should return a random quote', () => {
            global.Math.random = jest.fn().mockImplementation(() => 1)
            global.Math.floor = jest.fn().mockImplementation(() => 0)
            expect(wrapper.instance().handleRandomQuote()).toEqual('Don’t worry about Master Luke. I’m sure he’ll be all right. He’s quite clever, you know… for a human being.')
        });
    });

});


import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('Movie Card', () => {
    let wrapper;
    let mockChangeSelectedMovie = jest.fn()
    it('should match the snapshot with all the data passed in correctly', () => {
        wrapper =
            shallow(<MovieCard
                title={''}
                episode={3}
                img={''}
                key={3}
                releaseYear={''}
                characters={[]}
                changeSelectedMovie={mockChangeSelectedMovie}
                id={3} />)
        expect(wrapper).toMatchSnapshot();
    });

    it('should fire changeSelectedMovie when the View Characters Button is clicked', () => {
        let mockEvent = {
            target: {
                name: 'id',
                value: 3
            }
        }
        wrapper.find('.characters__button').simulate('onClick');
        expect(mockChangeSelectedMovie(mockEvent, 2)).toHaveBeenCalled();
    });
});
