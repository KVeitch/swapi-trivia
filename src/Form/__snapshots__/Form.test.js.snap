// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="blackout"
>
  <form>
    <div
      className="sidebar"
    >
      <h1>
        Star Wars Trivia
      </h1>
      <p
        className="hidden error"
      >
        <img
          alt="line drawing of R2D2"
          className="error-icon"
          src="003-robot.svg"
        />
        Please fill out all fields
      </p>
      <img
        alt="star-wars-icon"
        className="logo"
        src="https://image.flaticon.com/icons/svg/813/813466.svg"
      />
      <input
        className="input__name"
        name="name"
        onChange={[Function]}
        placeholder="What do you prefer to be called?"
        type="text"
        value=""
      />
      <input
        className="input__quote"
        name="quote"
        onChange={[Function]}
        placeholder="What is your favorite Star Wars quote?"
        type="text"
        value=""
      />
      <div
        className="selection__menu"
      >
        <h3>
          How Jedi Are You?
        </h3>
        <select
          name="ranking"
          onChange={[Function]}
          value="Nerf Herder"
        >
          <option
            className="option"
            value="Nerf Herder"
          >
            Nerf Herder
          </option>
          <option
            className="option"
            value="Padawan"
          >
            Padawan
          </option>
          <option
            className="option"
            value="Jedi Knight"
          >
            Jedi Knight
          </option>
          <option
            className="option"
            value="Jedi Master"
          >
            Jedi Master
          </option>
        </select>
      </div>
      <div
        className="btn__submit"
        onClick={[Function]}
        type="button"
      >
        Submit
      </div>
    </div>
    <div
      className="random__quote"
    >
      <p
        className="quote__text"
      >
        Don’t worry about Master Luke. I’m sure he’ll be all right. He’s quite clever, you know… for a human being.
      </p>
      <div
        className="quote__submit"
        onClick={[Function]}
        type="button"
      >
        Insults by C3P0
      </div>
    </div>
  </form>
</div>
`;
